docs/upgrade-notes/migrate-tea-standalone.md

```diff
@@ -0,0 +1,191 @@
+---
+id: migrate_tea_standalone
+title: Migrate TEA deployment to standalone module
+hide_title: false
+---
+
+## Background
+
+> This document is only relevant for upgrades of Cumulus from versions < 3.x.x to versions > 3.x.x
+
+Previous versions of Cumulus included deployment of the Thin Egress App (TEA) by default in the `distribution` module. As a result, Cumulus users who wanted to deploy a new version of TEA to wait on a new release of Cumulus that incorporated that release.
+
+In order to give Cumulus users the flexibility to deploy newer versions of TEA whenever they want, deployment of TEA has been removed from the `distribution` module and **Cumulus users must now add the TEA module to their deployment**. [Guidance on integrating the TEA module to your deployment is provided](deployment/thin-egress-app.md), or you can refer to [Cumulus core example deployment code for the `thin_egress_app` module](https://github.com/nasa/cumulus/blob/master/example/cumulus-tf/main.tf).
+
+By default, when upgrading Cumulus and moving from TEA deployed via the `distribution` module to deployed as a separate module, your API gateway for TEA would be destroyed and re-created, which could cause outages for any Cloudfront endpoints pointing at that API gateway.
+
+These instructions outline how to modify your state to preserve your existing Thin Egress App (TEA) API gateway when upgrading Cumulus and moving deployment of TEA to a standalone module. **If you do not care about preserving your API gateway for TEA when upgrading your Cumulus deployment, you can skip these instructions.**
+
+## Prerequisites
+
+- You [**must have bucket versioning enabled** for the Terraform state bucket used by your `cumulus` deployment](../deployment/terraform-best-practices#enable-bucket-versioning)
+
+## Notes about state management
+
+These instructions will involve manipulating your Terraform state via `terraform state mv` commands. These operations are **extremely dangerous**, since a mistake in editing your Terraform state can leave your stack in a corrupted state where deployment may be impossible or may result in unanticipated resource deletion.
+
+Since bucket versioning preserves a separate version of your state file each time it is written, and the Terraform state modification commands overwrite the state file, we can mitigate the risk of these operations by downloading the most recent state file before starting the upgrade process. Then, if anything goes wrong during the upgrade, we can restore that previous state version. Guidance on how to perform both operations is provided below.
+
+### Download your most recent state version
+
+Get the version ID of your most recent cumulus deployment state file, replacing `BUCKET` and `KEY` with the correct values from `cumulus-tf/terraform.tf`:
```

Comment:
Question: isn't the 'most recent state version' always grabbed via an `aws s3 cp` command?      I'm sure I'm missing something obvious, but why explicitly use s3api/get-object?

Reply:
Yeah... it is. I wasn't thinking clearly when I wrote this I guess. I'll update